### PR TITLE
Update specifications,roles and Best practices.

### PR DESCRIPTION
**Pull request by KiranVedantham**

### CODE REVIEW

File: mongodb.html.md.erb [hunk 1]
```diff
@@ -40,3 +40,29 @@ To connect to a running MongoDB instance with your local development tools, you
...
 
 Cloud Foundry: [Spring Music Example](https://github.com/cloudfoundry-samples/spring-music)
+
+## <a id='specifications'></a> Specifications
+
+Our [MongoDB](https://www.mongodb.com/community) offering consists of 3 Kubernetes containers configured as a replica set.
+
+## <a id='roles'></a> Roles
+
+You will receive the role `clusterMonitor` on admin db , `readWrite` and `dbAdmin` roles on your DB.
```
Comment (mastertinner) [inline mongodb.html.md.erb]: Must be "You will receive the role `clusterMonitor` on the admin DB, and the roles `readWrite` and `dbAdmin` on your DB."

File: mongodb.html.md.erb [hunk 2]
```diff
@@ -40,3 +40,29 @@ To connect to a running MongoDB instance with your local development tools, you
...
+
+## <a id='bestpractices'></a> Best Practices for Developers
+
+- Write concern 2
+
+  Please read [Replica Set Read and Write Semantics](https://docs.mongodb.com/manual/applications/replication). We recommend a "Write Concern" of `2` in order to ensure data consistency at all times.
+
+- Use an application driver maintained by the MongoDB team
+
+  It’s critical to find a driver that [is well-maintained](https://docs.mongodb.com/ecosystem/drivers/) and stays up to date with the latest MongoDB features.
```
Comment (mastertinner) [inline mongodb.html.md.erb]: "well-maintained" must be "well maintained"

File: mongodb.html.md.erb [hunk 3]
```diff
@@ -40,3 +40,29 @@ To connect to a running MongoDB instance with your local development tools, you
...
+
+- Index early and often
+
+  Without question, the most common database performance issue is improper indexing (or a complete lack thereof). Have a look at [Indexes docs](https://docs.mongodb.com/manual/indexes/).
```
Comment (mastertinner) [inline mongodb.html.md.erb]: "Have a look at [Indexes docs]" must be "Have a look at the [Indexes manual]"

**Comment by red-trela** [comment]
@mastertinner FYI, this should only be reviewed at this time, not yet merged - some of the changes are only true for the upcoming kubernetes-based version of the mongodb service.

**Comment by mastertinner** [comment]
Thanks, @red-trela. Please write me a comment here when it's ready.

**Pull request merged**